src/shared/components/app/navbar.tsx

```diff
@@ -44,14 +42,24 @@ function handleLogOut(i: Navbar) {
 }
 
 export class Navbar extends Component<NavbarProps, NavbarState> {
-  state: NavbarState = {
-    unreadInboxCountRes: { state: "empty" },
-    unreadReportCountRes: { state: "empty" },
-    unreadApplicationCountRes: { state: "empty" },
-  };
   collapseButtonRef = createRef<HTMLButtonElement>();
   mobileMenuRef = createRef<HTMLDivElement>();
 
+  state: NavbarState = {
+    unreadInboxCount: 0,
+    unreadReportCount: 0,
+    unreadApplicationCount: 0,
+  };
+
+  updateUnreads = (service: UnreadCounterService) => {
+    this.setState({
+      ...this.state,
+      unreadInboxCount: service.unreadInboxCount,
+      unreadReportCount: service.unreadReportCount,
+      unreadApplicationCount: service.unreadApplicationCount,
+    });
```

Comment:
No need to change if you don't want to, but this can be written more concisely like so:
```suggestion
  updateUnreads = ({
      unreadInboxCount,
      unreadReportCount,
      unreadApplicationCount
  }: UnreadCounterService) => {
    this.setState({
      ...this.state,
      unreadInboxCount,
      unreadReportCount,
      unreadApplicationCount,
    });
```